scheduler/cluster_scheduler.ml

```diff
@@ -151,28 +154,54 @@ let registration_services t =
 
 let pp_pool_name f (name, _) = Fmt.string f name
 
-let submission_service t =
+let submission_service ~validate ~sturdy_ref t client_id =
   let submit ~pool ~urgent descr =
-    match String.Map.find_opt pool t.pools with
-    | None ->
-      let msg = Fmt.strf "Pool ID %S not one of @[<h>{%a}@]" pool (String.Map.pp ~sep:Fmt.comma pp_pool_name) t.pools in
-      Capability.broken (Capnp_rpc.Exception.v msg)
-    | Some pool ->
-      Pool_api.submit ~urgent pool descr
+    match validate () with
+    | false -> Capability.broken (Capnp_rpc.Exception.v "Access has been revoked")
```

Comment:
Probably paranoid, but there's no need to leak the information in the error that this is a revoked capability is there? Just denying it as an invalid cap instead wouldn't share any info with an attacker.

Reply:
If we get here, the client has already successfully connected while the cap was valid, so they'll know it used to work.